pango/src/attr_iterator.rs

```diff
@@ -1,35 +1,220 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
-use crate::AttrIterator;
+use crate::AttrType;
 use crate::Attribute;
 use crate::FontDescription;
 use crate::Language;
 use glib::translate::*;
 
+use std::marker::PhantomData;
+use std::mem;
 use std::ptr;
 
-impl AttrIterator {
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct AttrIterator<'list> {
+    ptr: ptr::NonNull<ffi::PangoAttrIterator>,
+    list: PhantomData<&'list crate::AttrList>,
+}
+
+impl<'list> Clone for AttrIterator<'list> {
+    #[inline]
+    fn clone(&self) -> Self {
+        let ptr = unsafe {
+            ptr::NonNull::new_unchecked(ffi::pango_attr_iterator_copy(self.ptr.as_ptr()))
+        };
+        Self {
+            ptr,
+            list: PhantomData,
+        }
+    }
+}
+
+impl<'list> Drop for AttrIterator<'list> {
+    #[inline]
+    fn drop(&mut self) {
+        unsafe {
+            ffi::pango_attr_iterator_destroy(self.ptr.as_ptr());
+        }
+    }
+}
+
+#[cfg(any(feature = "v1_44", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v1_44")))]
+impl<'list> glib::StaticType for AttrIterator<'list> {
+    #[inline]
+    fn static_type() -> glib::Type {
+        unsafe { from_glib(ffi::pango_attr_iterator_get_type()) }
+    }
+}
+
+impl<'list> AttrIterator<'list> {
```

Comment:
I assume these methods are all just copy&paste of the autogenerated code plus adding the lifetime?

Reply:
Yes, except `font` was rewritten because that one was just broken